src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -67,34 +71,61 @@ abstract class AbstractCategoryType extends TranslatorAwareType
      */
     private $configuration;
 
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param array $customerGroupChoices
      * @param FeatureInterface $multistoreFeature
      * @param ConfigurationInterface $configuration
+     * @param Router $router
+     * @param CommandBusInterface $queryBus
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
-        ConfigurationInterface $configuration
+        ConfigurationInterface $configuration,
+        Router $router,
+        CommandBusInterface $queryBus
     ) {
         parent::__construct($translator, $locales);
 
         $this->customerGroupChoices = $customerGroupChoices;
         $this->multistoreFeature = $multistoreFeature;
         $this->configuration = $configuration;
+        $this->router = $router;
+        $this->queryBus = $queryBus;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $disableMenuThumbnailsUpload = false;
+        if (null !== $options['id_category']) {
+            $categoryId = $options['id_category'];
+            /** @var EditableCategory $editableCategory */
+            $editableCategory = $this->queryBus->handle(new GetCategoryForEditing((int) $categoryId));
```

Comment:
In theory, Queries are meant to be used in only 1 place. `GetCategoryForEditing` is a Query that fetches and return all the data needed for merchant to use an interface aimed at editing the category.

I think we should rather fetch this `canContainMoreMenuThumbnails` by another way:
- we can either use a Read Model like a DataProvider to fetch it
- or we can use the `$options` array to pass this data (see below)

In Controller, we already perform `$this->queryBus->handle(new GetCategoryForEditing((int) $categoryId));` . Then we could extract `canContainMoreMenuThumbnails` and inject it into the `$options` to be given to this form.

Reply:
Actually it completely make sense as we already go into this `if` following an option.

So `$options['disableMenuThumbnailsUpload']` looks good to me

Reply:
Adding is to options is great idea! Thanks, will do that.